components/HomeHeader.vue

```diff
@@ -124,50 +131,73 @@ export default Vue.extend({
     this.items = (
       await this.$api.userLibrary.getLatestMedia({
         userId: this.$auth.user.Id,
-        limit: 10,
+        limit: this.pages,
         fields: [ItemFields.Overview],
         enableImageTypes: [ImageType.Backdrop, ImageType.Logo],
         imageTypeLimit: 1
       })
     ).data;
-  },
-  methods: {
-    getBackdrop(item: BaseItemDto): string {
-      if (item.Type === 'Episode') {
-        return this.getImageUrlForElement(ImageType.Backdrop, {
-          itemId: item.SeriesId
-        });
-      } else if (item.Type === 'MusicAlbum') {
-        return this.getImageUrlForElement(ImageType.Backdrop, {
-          itemId: item.AlbumArtists?.[0].Id
-        });
+
+    for (const [key, i] of this.items.entries()) {
+      let id: string;
+      if (i.Type === 'Episode' && i.SeriesId) {
+        id = i?.SeriesId as string;
+      } else if (i.Type === 'MusicAlbum') {
+        id = i?.AlbumArtists?.[0].Id as string;
+      } else if (i.ParentLogoItemId) {
+        id = i?.ParentLogoItemId as string;
       } else {
-        return this.getImageUrlForElement(ImageType.Backdrop, {
-          itemId: item.Id
-        });
+        continue;
       }
-    },
-    getLogo(item: BaseItemDto): string {
-      if (item.Type === 'Episode' && item.SeriesId) {
-        return this.getImageUrlForElement(ImageType.Logo, {
-          itemId: item.SeriesId
-        });
-      } else if (item.ParentLogoItemId) {
-        return this.getImageUrlForElement(ImageType.Logo, {
-          itemId: item.ParentLogoItemId as string
-        });
-      } else {
-        return this.getImageUrlForElement(ImageType.Logo, {
-          item
-        });
+
+      const itemData = (
+        await this.$api.userLibrary.getItem({
+          userId: this.$auth.user.Id,
+          itemId: id
+        })
+      ).data;
+
+      this.relatedItems[key] = itemData;
+    }
+    this.loaded = true;
+  },
+  methods: {
+    getRelatedItem(item: BaseItemDto): BaseItemDto {
+      const rItem = this.relatedItems[this.items.indexOf(item)];
+      if (!rItem) {
+        return item;
       }
+      return rItem;
     },
     getOverview(item: BaseItemDto): string {
       if (item.Overview) {
         return this.sanitizeHtml(item.Overview);
       } else {
         return '';
       }
+    },
+    getLogo(item: BaseItemDto): string {
+      const relatedItem = this.getRelatedItem(item);
+      return this.getImageUrlForElement(ImageType.Logo, {
+        itemId: relatedItem.Id
+      });
+    },
+    onSlideChange(): void {
```

Comment:
```suggestion
    onSlideChange(swiperInstance: Swiper): void {
```

You already get the Swiper instance on all events emitted by Swiper itself, and the Vue library only passes them through. See https://swiperjs.com/api/#events

Reply:
@MrTimscampi ``swiperInstance`` returns ``undefined``